python/lsst/ip/isr/isrTask.py

```diff
@@ -1702,10 +1714,34 @@ def run(self, ccdExposure, *, camera=None, bias=None, linearizer=None,
 
         self.roughZeroPoint(ccdExposure)
 
+        # An extra config validation logic that checks for compatibility
+        # among the config options for amp offset correction.
+        if self.config.doApplyAmpOffset and not self.config.doMeasureAmpOffset:
+            raise RuntimeError("doApplyAmpOffset requires doMeasureAmpOffset to be True.")
+
+        # Measure amp offset corrections within the CCD.
+        if self.config.doMeasureAmpOffset:
+            ampOffsetConfig = AmpOffsetConfig()
```

Comment:
This should be done automatically if `self.ampOffset` is a subtask.  I think this also confuses the config that's written to disk, as it doesn't know about these changes.